bindings/java/src/org/sleuthkit/datamodel/blackboardutils/CommunicationArtifactsHelper.java

```diff
@@ -0,0 +1,719 @@
+
+			// post artifact 
+			getSleuthkitCase().getBlackboard().postArtifact(contactArtifact, getModuleName());
+		} catch (TskCoreException ex) {
+			logger.log(Level.SEVERE, "Unable to add contact artifact", ex); //NON-NLS
+			return null;
+		} catch (Blackboard.BlackboardException ex) {
+			logger.log(Level.SEVERE, String.format("Unable to post artifact %s", ((contactArtifact != null) ? contactArtifact.getArtifactID() : "")), ex);  //NON-NLS
+		}
+
+		return contactArtifact;
+	}
+
+	/**
+	 * Creates an account file instance associated with the DB file.
+	 *
+	 * @param accountType     type of account to create
+	 * @param accountUniqueID unique id for the account
+	 *
+	 * @return account instance created
+	 *
+	 * @throws TskCoreException
+	 */
+	private AccountFileInstance createAccountInstance(Account.Type accountType, String accountUniqueID) throws TskCoreException {
+		return getSleuthkitCase().getCommunicationsManager().createAccountFileInstance(accountType, accountUniqueID, getModuleName(), getAbstractFile());
+	}
+
+	/**
+	 * Adds a relations between the two specified account instances.
+	 *
+	 * @param selfAccount      device owner account
+	 * @param otherAccount     other account
+	 * @param sourceArtifact   artifact from which relationship is derived.
+	 * @param relationshipType type of relationship
+	 * @param dateTime         date/time of relationship
+	 */
+	private void addRelationship(AccountFileInstance selfAccountInstance, AccountFileInstance otherAccountInstance,
+			BlackboardArtifact sourceArtifact, Relationship.Type relationshipType, long dateTime) {
+		try {
+			if (selfAccountInstance.getAccount() != otherAccountInstance.getAccount()) {
+				getSleuthkitCase().getCommunicationsManager().addRelationships(selfAccountInstance,
+						Collections.singletonList(otherAccountInstance), sourceArtifact, relationshipType, dateTime);
+			}
+		} catch (TskCoreException | TskDataException ex) {
+			logger.log(Level.SEVERE, String.format("Unable to add relationship between account %s and account %s", selfAccountInstance.toString(), otherAccountInstance.toString()), ex); //NON-NLS
+		}
+	}
+
+	/**
+	 * Adds a TSK_MESSAGE artifact.
+	 *
+	 * Also creates an account instance for the sender/receiver, and creates a
+	 * relationship between the self account and the sender/receiver account.
+	 *
+	 * @param messageType message type, may be empty or null
+	 * @param direction   message direction, UNKNOWN if not available
+	 * @param fromAddress sender address, may be null
+	 * @param toAddress   recipient address, may be null
+	 * @param dateTime    date/time of message, 0 if not available
+	 * @param readStatus  message read or not, UNKNOWN if not available
+	 * @param subject     message subject, may be empty or null
+	 * @param messageText message body, may be empty or null
+	 * @param threadId,   message thread id, may be empty or null
+	 *
```

Comment:
The parameter documentation here strikes me as odd. Why can everything be UNKNOWN, empty, or null? What are the minimum attributes for a message artifact?